src/meshes/t8code_mesh.jl

```diff
@@ -52,7 +51,7 @@ mutable struct T8codeMesh{NDIMS, RealT <: Real, IsParallel, NDIMSP2, NNODES} <:
             # further down. However, this might cause a pile-up of `mesh`
             # objects during long-running sessions.
             if !MPI.Finalized()
-                trixi_t8_unref_forest(mesh.forest)
+                t8_forest_unref(Ref(mesh.forest))
```

Comment:
Why this change? Did T8code.jl introduce the new functions some time ago - after the previous version using `trixi_t8_unref_forest` was written? If so, do our compat settings in Project.toml reflect this correctly?
The same question holds for other functions removed from src/auxiliary/t8code.jl

Reply:
These were just helper resp. wrapper functions. Since `trixi_t8_unref_forest` is only used just twice in the whole Trixi source code I decided to just remove it and call `t8_forest_unref` directly. The API of t8code or T8code.jl did not change at all.

Just cleaning the code in the PR a bit.